### PR TITLE
Don't send RR or RM when peeking at a room

### PR DESCRIPTION
**Pull request by turt2live**
Fixes vector-im/riot-web#696

### CODE REVIEW

File: src/components/structures/RoomView.js [hunk 1]
```diff
@@ -1728,8 +1731,8 @@ module.exports = React.createClass({
             <TimelinePanel ref={this._gatherTimelinePanelRef}
                 timelineSet={this.state.room.getUnfilteredTimelineSet()}
                 showReadReceipts={!UserSettingsStore.getSyncedSetting('hideReadReceipts', false)}
-                manageReadReceipts={true}
-                manageReadMarkers={true}
+                manageReadReceipts={!this.state.isPeeking}
```
Comment (t3chguy) [inline src/components/structures/RoomView.js]: doesn't this also mean that you won't be able to see read receipts whilst peeking? I know the additional prop showReadReceipt is there but I think it requires manageReadReceipts=true

File: src/components/structures/RoomView.js [hunk 2]
```diff
@@ -266,6 +267,7 @@ module.exports = React.createClass({
                 console.log("Attempting to peek into room %s", roomId);
                 this.setState({
                     peekLoading: true,
+                    isPeeking: true, // this will change to false if peeking fails
```
Comment (lukebarnard1) [inline src/components/structures/RoomView.js]: I don't think this comment is true

File: src/components/structures/RoomView.js (hunk 1 above)
Comment (turt2live) [reply src/components/structures/RoomView.js]: I'll double check, but in my testing I don't remember it being failing.

File: src/components/structures/RoomView.js (hunk 2 above)
Comment (turt2live) [reply src/components/structures/RoomView.js]: @lukebarnard1 good catch, fixed.

File: src/components/structures/RoomView.js (hunk 1 above)
Comment (turt2live) [reply src/components/structures/RoomView.js]: can confirm: read receipts still visible for other people.

File: src/components/structures/RoomView.js (hunk 2 above)
Comment (lukebarnard1) [reply src/components/structures/RoomView.js]: TBH, for symmetry I'd move `isPeeking: true` to line 275

File: src/components/structures/RoomView.js [hunk 3]
```diff
@@ -281,6 +283,7 @@ module.exports = React.createClass({
                         // This is fine: the room just isn't peekable (we assume).
                         this.setState({
                             peekLoading: false,
+                            isPeeking: false,
```
Comment (lukebarnard1) [inline src/components/structures/RoomView.js]: I think we also want `isPeeking: false` in the general case of an error when trying to peek (e.g. internal server error), not that it matters hugely, more for consistency.

File: src/components/structures/RoomView.js (hunk 2 above)
Comment (turt2live) [reply src/components/structures/RoomView.js]: I started by doing that, and it effectively caused RRs to be sent on initial peek

File: src/components/structures/RoomView.js (hunk 3 above)
Comment (lukebarnard1) [reply src/components/structures/RoomView.js]: (also note my comment [here](https://github.com/matrix-org/matrix-react-sdk/pull/1463#discussion_r144285513))

File: src/components/structures/RoomView.js (hunk 2 above)
Comment (lukebarnard1) [reply src/components/structures/RoomView.js]: oh ok, fair enough!

File: src/components/structures/RoomView.js (hunk 3 above)
Comment (turt2live) [reply src/components/structures/RoomView.js]: fixed

**Pull request merged**